### DISCUSSION TITLE
How to perform server side admin operations with service_role key?

### DISCUSSION THREAD

**Discussion by oliver-tunebat** [supabase/supabase#9914] [Questions] [answered]
Using @supabase/auth-helpers-nextjs 0.4.1. I need to have an end point that deletes a user. The docs mention that a service_role key is required, but I don't see any documentation for creating a client with a service_role key. It looks like all clients are automatically created only using the anon_key environment variable.

**Comment by dshukertjr** [comment, marked as answer]
You can simply initialize a Supabase client with your service role key instead of the anon key. 

```typescript
const adminSupabase = createClient('https://xyzcompany.supabase.co', 'my_service_role_key')
```
